File: CS/ColumnBindingExample/ColumnBindingHelper.cs
using System.Windows;
using System.Windows.Data;

namespace ColumnBindingExample {
    public class ColumnBindingHelper {
        public static readonly DependencyProperty BindingPathProperty = DependencyProperty.RegisterAttached("BindingPath", typeof(string), typeof(ColumnBindingHelper), new PropertyMetadata(null, OnBindingPathChanged));

        private static void OnBindingPathChanged(DependencyObject d, DependencyPropertyChangedEventArgs e) {
            var column = d as DevExpress.Xpf.Grid.GridColumn;
            if (column == null || e.NewValue == null)
                return;
            column.Binding = new Binding(e.NewValue.ToString());
        }

        public static string GetBindingPath(DependencyObject obj) {
            return (string)obj.GetValue(BindingPathProperty);
        }

        public static void SetBindingPath(DependencyObject obj, string value) {
            obj.SetValue(BindingPathProperty, value);
        }
    }
}

File: VB/ColumnBindingExample/ColumnBindingHelper.vb
Imports System.Windows
Imports System.Windows.Data

Namespace ColumnBindingExample
    Public Class ColumnBindingHelper
        Public Shared ReadOnly BindingPathProperty As DependencyProperty = DependencyProperty.RegisterAttached("BindingPath", GetType(String), GetType(ColumnBindingHelper), New PropertyMetadata(Nothing, AddressOf OnBindingPathChanged))

        Private Shared Sub OnBindingPathChanged(ByVal d As DependencyObject, ByVal e As DependencyPropertyChangedEventArgs)
            Dim column = TryCast(d, DevExpress.Xpf.Grid.GridColumn)
            If column Is Nothing OrElse e.NewValue Is Nothing Then
                Return
            End If
            column.Binding = New Binding(e.NewValue.ToString())
        End Sub

        Public Shared Function GetBindingPath(ByVal obj As DependencyObject) As String
            Return DirectCast(obj.GetValue(BindingPathProperty), String)
        End Function

        Public Shared Sub SetBindingPath(ByVal obj As DependencyObject, ByVal value As String)
            obj.SetValue(BindingPathProperty, value)
        End Sub
    End Class
End Namespace
